### PR TITLE
tool: ceph-kvstore-tool doesn't umount BlueStore properly

### PR DESCRIPTION
**Pull request by Liuchang0812**
Fixes: http://tracker.ceph.com/issues/21625

Signed-off-by: Chang Liu <liuchang0812@gmail.com>

### CODE REVIEW

File: src/tools/ceph_kvstore_tool.cc [hunk 1]
```diff
@@ -64,7 +70,18 @@ class StoreTool
 	exit(1);
       }
     }
-    db.reset(db_ptr);
+    db = db_ptr;
+  }
+
+  ~StoreTool() {
+    if (bluestore) {
+      assert(0 == bluestore->umount());
```
Comment (liewegas) [inline src/tools/ceph_kvstore_tool.cc]: please move the call with side effects out of the assert, in case we compile out assert() in the future.

Comment (Liuchang0812) [reply src/tools/ceph_kvstore_tool.cc]: thank you Sage. done, remove assert here.

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-10-03_21:58:15-rados-wip-sage-testing-2017-10-03-1358-distro-basic-smithi/

**Comment by liewegas** [comment]
@Liuchang0812 do you mind writing a test case that reproduces the original issue?  can probably put it in qa/standalone.. maybe a separate dir for tool tests that don't need an actual cluster?

**Comment by Liuchang0812** [comment]
I will do it. we need a bluestore dir, so standalone seems is better?

**Comment by liewegas** [comment]
ceph-objectstore-tool --op mkfs might be helpful.

**Comment by Liuchang0812** [comment]
@liewegas I have wrote a test case for ceph-kvstore-tool, see https://github.com/ceph/ceph/pull/16745/commits/47525136a0d9199b5e6491cc9209b54608e19bc4

**Comment by liewegas** [comment]
Looks good.. is that commit part of another PR?  We also need to add a .yaml file in qa/suites/rados/standalone-nomgrs/... somewhere so that it gets run as part of the test suite.

**Comment by Liuchang0812** [comment]
Looks good.. is that commit part of another PR? 

yeah, PR: https://github.com/ceph/ceph/pull/16745

   We also need to add a .yaml file in qa/suites/rados/standalone-nomgrs/... somewhere so that it gets run as part of the test suite.

I put it in `qa/workunits` since it doesn't need a standalone cluster. it seems that we will run it automatically. not sure....I'm learning how to write a yaml file.

**Pull request merged**